### DISCUSSION TITLE
Git Actions 를 돌릴 때 /bin/sh에 ^M이 포함되어 돌아가지않는 문제

### DISCUSSION THREAD

**Discussion by numerical43** [meta-mingles/metamingle-server#55] [Troubleshooting]
## 문제 상황 (오류 메세지)
pr을 올린 뒤 git actions가 돌아가다가 에러가 뜨면서 빌드 테스트가 완료되지않는 상황 발생

```
Run ./gradlew --info test
/home/runner/work/_temp/5916e610-a480-4315-bea9-5952a1728cba.sh: ./gradlew: /bin/sh^M: bad interpreter: No such file or directory
Error: Process completed with exit code 126.
```

<br>

## 문제 파악
스크립트의 첫 번째 줄인 shebang (#!) 다음에 나오는 경로가 `/bin/sh^M`으로 해석되었는데, 원래` /bin/sh`이라는 경로에 `^M`이라는 문자가 합쳐서` /bin/sh^M` 이라는 경로로 인식되고 있습니다. 여기서 `^M`은 DOS 스타일의 줄바꿈 문자를 표현하는 방식입니다.

오류 메세지를 해석해보면 `gradlew` 파일이 Windows 환경에서 작성되어 DOS 스타일의 `줄바꿈 문자(\r\n)`를 포함하고 있고, 해당 파일을 현재 Linux 환경에서 실행하려고하면서 문제가 발생하고 있습니다. (git actions의 환경은 ubuntu-latest이기 때문에)

<br>

## 문제 해결
아래 코드처럼 `sed` 명령어를 사용하여 gradlew 파일 내의` \r `문자를 제거해서 해결할 수 있었습니다.🪄
``` yaml
- name: Add permission
  run: 
    chmod +x gradlew;
    sed -i 's/\r$//' gradlew
```

**Comment by silverpoodle** [comment]
확인했습니다. 참고하겠습니다!
